sos/plugins/openvswitch.py

```diff
@@ -42,6 +43,21 @@ def setup(self):
             "/var/run/openvswitch/ovsdb-server.pid",
             "/var/run/openvswitch/ovs-vswitchd.pid"
         ])
+        # Collect ovs-vswitchd's pmd thread details in procfs.
+        pid = open("/var/run/openvswitch/ovs-vswitchd.pid").read()\
+            .splitlines()[0]
+        # List TID & thread command name.
+        ps_tidlist_cmd = "ps -p %s -Lo tid=,comm= " % pid
+        tid_list = filter(
+                re.compile(r'pmd').search,
+                self.call_ext_prog(ps_tidlist_cmd)['output'].splitlines())
```

Comment:
Same here: what if "ps" would fail (unprobable) or have no output (e.g. the process with given PID does not (further) exist)?

Reply:
Thanks for your inputs, Just to clarify , do you mean a stale pid file is present but the process is not running ( killed perhaps) ?

Reply:
Taking my comment back here - even in such situations when "ps" command fails or returns nothing, the worst what we get is empty `tid_list`. And that is handled below well.

Reply:
If you already have a PID from a pidfile this seems like an unnecessarily complex way of obtaining the TIDs. Something like this should be adequate:

```python
    proc_files = ["environ", "cgroup", "maps", "numa_maps", "limits", "status"]
    try:
        pid = open("/var/run/openvswitch/ovs-vswitchd.pid").read().splitlines()[0]
        self.add_copy_spec(["/proc/%s/task/*/%s" % (pid, pf) for pf in proc_files])
    except:
         self._log_error("Could not read ovs-vswitch PID")
```